### PR TITLE
adding the ultradns api functions for acme.sh as well as appropriate …

### PR DESCRIPTION
**Pull request by mianosm**
…readme markdown

### CODE REVIEW

File: dnsapi/README.md [hunk 1]
```diff
@@ -1259,6 +1259,22 @@ acme.sh --issue --dns dns_online -d example.com -d www.example.com
 
 `ONLINE_API_KEY` will be saved in `~/.acme.sh/account.conf` and will be reused when needed.
 
+## 66. Use UltraDNS API
+
```
Comment (Neilpang) [inline dnsapi/README.md]: can you please share the ultradns website url here ?

File: dnsapi/dns_ultra.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+  _debug "Getting TXT records"
+  _ultra_rest GET "zones/${_domain_id}/rrsets?q=kind:RECORDS+owner:${_sub_domain}"
+
+  if ! printf "%s" "$response" | grep \"resultInfo\" >/dev/null; then
+    _err "There was an error in obtaining the resource records for ${_domain_id}"
+    return 1
+  fi
+
+  count=$(printf "%s\n" "$response" | _egrep_o "\"returnedCount\":[^,]*" | cut -d: -f2 | cut -d'}' -f1)
```
Comment (Neilpang) [inline dnsapi/dns_ultra.sh]: please use `echo` here instead of `printf`

File: dnsapi/dns_ultra.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  if [ "$m" != "GET" ]; then
+    _debug data "${data}"
+    response="$(_post "${data}" "${ULTRA_API}"/"${ep}" "" "${m}")"
+  else
+    response="$(_get "$ULTRA_API/$ep")"
+  fi
+}
+
+_ultra_login() {
+  AUTH_TOKEN=$(curl -X POST --data \
```
Comment (Neilpang) [inline dnsapi/dns_ultra.sh]: do not use `curl`.
please use `_get()`  or `_post()` instead.

File: dnsapi/README.md (hunk 1 above)
Comment (mianosm) [reply dnsapi/README.md]: @Neilpang - I've added the URL for the UltraDNS portal in the dnsapi/README.md file.

File: dnsapi/dns_ultra.sh (hunk 2 above)
Comment (mianosm) [reply dnsapi/dns_ultra.sh]: Replaced the printf with an echo.

File: dnsapi/dns_ultra.sh (hunk 3 above)
Comment (mianosm) [reply dnsapi/dns_ultra.sh]: I've updated the _ultra_login function to utilize the _post() function in place of curl. Thanks for catching this mistake!

**Comment by Neilpang** [comment]
How to use ultradns service ?
Is it free ?  or you must register domain with them ?  or something else ?

**Comment by mianosm** [comment]
> How to use ultradns service ?
> Is it free ? or you must register domain with them ? or something else ?

The REST API user guide is unfortunately behind a login here: https://portal.ultradns.com/static/docs/REST-API_User_Guide.pdf

The DNS services are provided by Neustar, and you must be a paying customer, main page is here: 
https://www.security.neustar/lp/ultra20/index.html

**Comment by Neilpang** [comment]
> > How to use ultradns service ?
> > Is it free ? or you must register domain with them ? or something else ?
> 
> The REST API user guide is unfortunately behind a login here: https://portal.ultradns.com/static/docs/REST-API_User_Guide.pdf
> 
> The DNS services are provided by Neustar, and you must be a paying customer, main page is here:
> https://www.security.neustar/lp/ultra20/index.html


ok,  please update these comments in the readme file.

Thanks.

**Comment by mianosm** [comment]
> > > How to use ultradns service ?
> > > Is it free ? or you must register domain with them ? or something else ?
> > 
> > 
> > The REST API user guide is unfortunately behind a login here: https://portal.ultradns.com/static/docs/REST-API_User_Guide.pdf
> > The DNS services are provided by Neustar, and you must be a paying customer, main page is here:
> > https://www.security.neustar/lp/ultra20/index.html
> 
> ok, please update these comments in the readme file.
> 
> Thanks.

Thanks Neilpang! I've added the additional URLs and a brief insight on what/how UltraDNS does what they do.

**Comment by Neilpang** [comment]
please close this PR, and  push a new PR to dev branch

**Comment by mianosm** [comment]
I've updated the merge destination request, is that okay, or do you still want a new PR to dev branch?

**Comment by Neilpang** [comment]
sorry, I just updated the dev branch, please resolve the conflicts now, I think it will be ok.

Thanks.

**Pull request merged**